content/en/observability_pipelines/processors/remap_ocsf/custom_mapping_configuration_format.md

```diff
@@ -0,0 +1,418 @@
+| `matches: R`    | Matches if `value` is a string that matches the regular expression `R`.                        |
+| `not_matches: R`| Matches if `matches` above does not.                                                            |
+
+The table may also include a `default` value to be used if none of the comparisons succeed. If a `default` is not configured and none of the values match, the source value is passed through as is.
+
+{{< tabs >}}
+{{% tab "YAML example" %}}
+
+```yaml
+mapping:
+- dest: src_endpoint.os.type
+  source: client.userAgent.os
+  lookup:
+    table:
+    - contains: Windows
+      value: Windows
+    - equals: Linux
+      value: Linux
+    - matches: "^[Mm]ac *OS"
+      value: macOS
+    default: Unknown
+- dest: activity_name
+  source: createdDateTime
+  lookup:
+    table:
+    - equals_source: lastUpdatedDateTime
+      value: Create
+    default: Update
+```
+
+{{% /tab %}}
+{{% tab "JSON example" %}}
+
+```json
+{
+  "mapping": [
+    {
+      "dest": "src_endpoint.os.type",
+      "source": "client.userAgent.os",
+      "lookup": {
+        "table": [
+          {
+            "contains": "Windows",
+            "value": "Windows"
+          },
+          {
+            "equals": "Linux",
+            "value": "Linux"
+          },
+          {
+            "matches": "^[Mm]ac *OS",
+            "value": "macOS"
+          }
+        ],
+        "default": "Unknown"
+      }
+    },
+    {
+      "dest": "activity_name",
+      "source": "createdDateTime",
+      "lookup": {
+        "table": [
+          {
+            "equals_source": "lastUpdatedDateTime",
+            "value": "Create"
+          }
+        ],
+        "default": "Update"
+      }
+    }
+  ]
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### Mapping functions
+
+A function applies an operation to the value extracted from the source, before assigning the extracted value to the destination.
+
+### `concat_arrays`
+
+The `concat_arrays` function looks up a second `source` and, if it exists, concatenates it to the original source data. If a source is not an array, the source is converted into an array with a single element. The result is always an array.
+
+{{< tabs >}}
+{{% tab "YAML example" %}}
+
+```yaml
+- dest: metadata.labels
+  source: systemTags
+  function:
+    name: concat_arrays
+    source: customTags
+```
+
+{{% /tab %}}
+{{% tab "JSON example" %}}
+
+```json
+{
+  "dest": "metadata.labels",
+  "source": "systemTags",
+  "function": {
+    "name": "concat_arrays",
+    "source": "customTags"
+  }
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+
+### `reshape_array`
+
+The `reshape_array` function extracts data from a source array to create a new array of values. The function filters only array elements containing a field that matches a condition from the list in [Mapping lookup tables](#mapping-lookup-tables), and extracts another field into the output array.
```

Comment:
```suggestion
The `reshape_array` function extracts data from a source array to create a new array of values. The function filters array elements, selecting only those that contain a field matching a condition from the [Mapping lookup tables](#mapping-lookup-tables), and extracts another field into the output array.
```

Reply:
This might be because I'm not very familiar with OCSF or Observability Pipelines, but I'm not sure I understand what 'another field' refers to in the phrase 'and extracts another field into the output array.'"

Reply:
That is a reference to a field within elements of the array. i.e. if the array contains `[{"a":1},{"a":2}]` then you could indicate you want to extract field `a` and it would result in `[1, 2]`.